sklearn/datasets/_base.py

```diff
@@ -280,6 +285,67 @@ def load_data(module_path, data_file_name):
     return data, target, target_names
 
 
+def load_gzip_compressed_csv_data(data_module, data_file_name, **kwargs):
+    """Loads data_file_name from data_module with importlib.resources and np.loadtxt
+
+    1) Open resource file with `importlib.resources.open_binary`
+    2) Decompress file obj with `gzip.open`
+    3) Load decompressed data with `np.loadtxt`
+
+    Parameters
+    ----------
+    data_module : string or module
+        module where data lives;
+
+    data_file_name : string
+        Name of gzip-compressed csv (.csv.gz) file to be loaded from
+        data_module/data_file_name. For example 'diabetes_data.csv.gz'.
+
+    **kwargs : dict
+        Keyword arguments to be passed to `np.loadtxt`;
+        e.g. delimiter=","
+
+    Returns
+    -------
+    data : Numpy array
+        A 2D array with each row representing one sample and each column
+            representing the features and/or target of a given sample.
+    """
+    with importlib_resources.open_binary(
+        data_module, data_file_name
+    ) as compressed_file:
+        # GH20297: The following handling of the `encoding` kwarg
+        # should follow that in `np.loadtxt`
+        gzip_encoding = None
+        if "encoding" in kwargs and kwargs["encoding"] != "bytes":
+            gzip_encoding = kwargs["encoding"]
+
+        compressed_file = gzip.open(compressed_file, mode="rt", encoding=gzip_encoding)
+        data = np.loadtxt(compressed_file, **kwargs)
+    return data
+
+
+def load_descr(descr_module, descr_file_name):
+    """Loads descr_file_name from descr_module with importlib.resources.
+
+    Parameters
+    ----------
+    descr_module : string or module
+        module where descr_file_name lives;
+
+    descr_file_name : string
+        Name of rst file to be loaded from
+        descr_module/descr_file_name. For example 'wine_data.rst'.
+
+    Returns
+    -------
+    string; content of descr_file_name
+    """
+    fdescr = importlib_resources.read_text(descr_module, descr_file_name)
+
+    return fdescr
```

Comment:
As far as I can see most of the descriptions are loaded from `DESCR_MODULE` so to avoid redundancy we could have:

```
load_descr(descr_file_name, descr_module=DESCR_MODULE):
```

that would make it possible to only import `load_descr` without having to import the `DESCR_MODULE` constant most of the time.

Reply:
For convenience you could add an optional `descr_file_name=None` / `descr_module=DESCR_MODULE` to `load_csv_data` and `load_gzip_compressed_csv_data` and when `descr_file_name` is not `None`, make those functions return the loaded description as as additional return value.

Reply:
> As far as I can see most of the descriptions are loaded from `DESCR_MODULE` so to avoid redundancy we could have:
> 
> ```
> load_descr(descr_file_name, descr_module=DESCR_MODULE):
> ```
> 
> that would make it possible to only import `load_descr` without having to import the `DESCR_MODULE` constant most of the time.

By extension, I am applying the same changes to `load_csv_data` and `load_gzip_compressed_csv_data` to avoid the same redundancy since these data are always loaded from `DATA_MODULE`.  
```
load_csv_data(data_file_name, data_module=DATA_MODULE)
```
And since as positional arguments to `importlib.resources` functions, the resource module always comes before the resource name, I am enforcing named arguments to avoid usage error.
```
load_csv_data(data_file_name, *, data_module=DATA_MODULE)
load_descr(descr_file_name, *, descr_module=DESCR_MODULE)
```
Lastly, combined with the suggestion to add optional `descr_file_name=None, descr_module=DESCR_MODULE`, we have the following function signature:
```
load_csv_data(data_file_name, *, data_module=DATA_MODULE, descr_file_name=None, descr_module=DESCR_MODULE)
```